src/snaphu/_check.py

```diff
@@ -70,6 +75,74 @@ def check_dataset_shapes(
             raise ValueError(errmsg)
 
 
+def check_dataset_sizes(
+    ntiles: tuple[int, int],
+    tile_overlap: int | tuple[int, int],
+    *,
+    regrow_conncomps: bool = True,
+    single_tile_reoptimize: bool = False,
+    **datasets: InputDataset | OutputDataset,
+) -> None:
+    """
+    Ensure that one or more datasets have shape that snaphu can handle.
+
+    Parameters
+    ----------
+    ntiles : (int, int)
+        Number of tiles used in each dimension.
+    tile_overlap: int or (int, int)
+        Overlap between tiles.
+    regrow_conncomps : bool
+        Whether to regrow connected components after tiled unwrapping.
+    single_tile_reoptimize: bool
+        Whether to use single tile reoptimization after tiled unwrapping.
+    **datasets : dict, optional
+        Datasets whose shape must be equal to `shape`. The name of each keyword argument
+        is used to format the error message in case of a size exception.
+
+    Raises
+    ------
+    ValueError
+        If any dataset had a too large shape.
+    TypeError
+        If the tile overlaps type is unknown.
+    """
+    if isinstance(tile_overlap, int):
+        y_overlap = x_overlap = tile_overlap
+    elif isinstance(tile_overlap, tuple):
+        # cannot unpack, tile_overlap shape is checked later
+        y_overlap = tile_overlap[0]
+        x_overlap = tile_overlap[1]
+
+    else:
+        msg = f"Unknown format tile overlaps: {tile_overlap}"
+        raise TypeError(msg)
+
+    for name, arr in datasets.items():
+        skip_tiling: bool = ntiles == (1, 1)
+        if regrow_conncomps or single_tile_reoptimize or skip_tiling:
+            # a single tile is input for snaphu
+            for size in arr.shape:
+                if size > LARGESHORT:
+                    msg = f"dataset {name} too large for snaphu, shape: {arr.shape}"
+                    raise ValueError(msg)
+            if skip_tiling:
+                return
+
+        # in case tile exceeds max array size
+        def _calc_tile_shape(array_len: int, num_tiles: int, overlap_len: int):
```

Comment:
Let's annotate the return type here as well.

```suggestion
        def _calc_tile_shape(array_len: int, num_tiles: int, overlap_len: int) -> int:
```